### ISSUE TITLE
Add number pseudo-type

### ISSUE THREAD

**Issue by zonuexe** [phpDocumentor/fig-standards#142]
`number` pseudo-type can be found in the PHP manual.
This keyword is defined in [PHP: Pseudo-types and variables used in this documentation - Manual](http://php.net/manual/en/language.pseudo-types.php).

PhpStorm seems to already implement this keyword as a pseudo-type.

I propose to add `number` into [Appendix A. Types](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#keyword).

**Comment by ashnazg** [comment]
With this being a recognized pseudo-type by the PHP manual, I'm inclined to agree.

**Comment by neuro159** [comment]
"PhpStorm seems to already implement this keyword as a pseudo-type."
First of all - I think we shall move in direction to having same types in php code and doc.

Anyway - as most of doc-related stuff we support - in the beginning I scanned a LOT of projects and used the stats. Number was there, especially in older code. So I just made it behave intuitively (like base class for int and float).

We usually only generate actual inferred type int or float if possible, and thus see a decline on "number" usage along years.

Still gives us some pain - PPL make classes called Number and more
https://youtrack.jetbrains.com/issue/WI-40675
was same with String and Object custom classes but nowadays very rare.

Would be great to have such type names reserved. 
May be "deprecated" == never generated by tool and treated in type computations as int|float.
